### PR TITLE
Add reconstruction method to ConstantVariable

### LINKED ISSUE #96517: Dynamo cannot reconstruct ConstantVariable
### 🐛 Describe the bug

repro:

```
import torch
from torch._dynamo.comptime import comptime
import contextlib
import logging

@torch._dynamo.optimize("eager")
def h():
    x = set()
    comptime.graph_break()
    x.add(1)
    return x

h()
```

I ran into this while debugging an unrelated error in dynamo runs on the PyTorch test suite. Feel free to treat with according priority.

This is not too hard to fix, I'm happy to advise.

### Versions

master

### PR DESCRIPTION
**Pull request by knwng**
Fixes #96517

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov @soumith @desertfire

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
voz this looks test bankruptcy affected

File: test/dynamo/test_repros.py [hunk 1]
```diff
@@ -3190,6 +3190,23 @@ def forward(self, param):
...
+        torch._dynamo.config.verbose = True
+
+        @torch._dynamo.optimize("eager")
+        def h():
+            x = set()
+            comptime.graph_break()
+            x.add(1)
+            return x
+
+        h()
```
Comment (voznesenskym) [inline test/dynamo/test_repros.py:3208]: Can you do this w/ compile counter?

**Review by voznesenskym — APPROVED** [review]

File: torch/_dynamo/variables/constant.py [hunk 2]
```diff
@@ -139,6 +139,9 @@ def call_hasattr(self, tx, name: str) -> "VariableTracker":
         result = hasattr(self.value, name)
         return variables.ConstantVariable(result).add_options(self)
 
+    def reconstruct(self, codegen):
+        return [codegen._create_load_const(self.value)]
```
Comment (voznesenskym) [inline torch/_dynamo/variables/constant.py:143]: use create_load_const, not the underscored one.

**Comment by voznesenskym** [comment]
Please ensure the tests pass - you need to see if the failures are real or not. Part of this involves double checking that create_load_const does the right thing here in the cases of these PRs.

**Pull request closed without merging**